plugins/checklist/assets/javascripts/lib/discourse-markdown/checklist.js

```diff
@@ -1,4 +1,4 @@
-const REGEX = /\[(\s?|x|X)\]/g;
+const REGEX = /\[( |x)?\]/gi;
```

Comment:
Technically not part of the fix, but we only allow a whitespace and either "x" or "X".